Update _CPlugin_Helper.cpp

Rini52 opened the pull request:
need client added in call to compile

TD-er committed:
[Playground] Move P168_ThermoOLED.ino to main repo

Still need to fix parsing template strings on screen.

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

TD-er committed:
Rename P107_ThermoOLED to P109

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

tonhuisman committed:
[P065] Add mode and repeat commands
From [a request in the forum](https://www.letscontrolit.com/forum/viewtopic.php?f=4&t=8521)
- mode,<0..3> to set playback mode: 0=Repeat, 1=Folder repeat, 2=Single repeat, 3=Random
- repeat,<0/1> enable/disable repeat play 0=Off, 1=On

TODO:
- [ ] Feedback from tester

tonhuisman committed:
[P065] Uncrustify and some minor source adjustments

tonhuisman committed:
[P065] Reviewer corrections

tonhuisman committed:
[P065] Minor optimization

tonhuisman committed:
[P065] Add documentation (partly migrated from the wiki)

TD-er committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

tonhuisman committed:
[P065] Add commands to command reference

tonhuisman committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

tonhuisman committed:
[P065] Add commands to command reference (cleanup)

tonhuisman committed:
[UI] Enable optional tooltip for most input controls

tonhuisman committed:
[UI] Add tooltip option for a few more input controls

tonhuisman committed:
[UI] Add conditional compilation to Tooltip feature

tonhuisman committed:
Merge branch 'mega' into feature/ui-add-optional-tooltip-for-input-controls

tonhuisman committed:
Merge branch 'mega' into feature/ui-add-optional-tooltip-for-input-controls

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

TD-er committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

TD-er committed:
Merge branch 'mega' into feature/Playground_P168_ThermoOLED

tonhuisman committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

tonhuisman committed:
Merge branch 'mega' of https://github.com/tonhuisman/ESPEasy-1 into feature/ui-add-optional-tooltip-for-input-controls

tonhuisman committed:
[UI] Rename `title` argument to `tooltip`

tonhuisman committed:
[UI] Fix manual merge typo

tonhuisman committed:
[Build] Include Trigonometric functions also in 1M Normal builds

tonhuisman committed:
[Doc] Correct function names for asin, acos, atan and friends

tonhuisman committed:
[Build] Keep trigonometric stuff out of MINIMAL_OTA builds

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/ui-add-optional-tooltip-for-input-controls

tonhuisman committed:
[UI] Correct manual merge error

tonhuisman committed:
[UI] Add extra definition of addSelector() with tooltip support

tonhuisman committed:
Merge branch 'mega' into bugfix/trigonometric-functions-in-normal-1M

Rini52 committed:
Update _CPlugin_Helper.cpp

need client

TD-er commented:
How do you compile it?
Sounds like you don't have the defines configured as how PlatformIO does.

TD-er committed:
Merge branch 'mega' into feature/ui-add-optional-tooltip-for-input-controls

TD-er committed:
[Tooltips] Fix merge issues

tonhuisman committed:
[P053] Add support for PMS5003ST, PMS2003 and PMS3003

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p053-add-support-for-pms5003st-pms2003-pms3003

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/ui-add-optional-tooltip-for-input-controls

TD-er committed:
[Cleanup] Uncrustify C014 Homie

TD-er committed:
[Cleanup] C014 Log entries using F()

tonhuisman committed:
Merge branch 'mega' into bugfix/trigonometric-functions-in-normal-1M

TD-er committed:
[C014] Cleanup calls to CPlugin_014_sendMQTTdevice not using F()

TD-er committed:
[C014] Cleanup calls to CPlugin_014_sendMQTTnode not using F()

TD-er committed:
Merge branch 'mega' into mega

TD-er committed:
[DS1307] Add support for external RTC clock DS1307

ToDo:

- [ ] Test
- [ ] Document

tonhuisman committed:
Merge branch 'mega' into feature/ui-add-optional-tooltip-for-input-controls

TD-er committed:
Merge pull request #3641 from tonhuisman/feature/ui-add-optional-tooltip-for-input-controls

[UI] Enable optional tooltip for most input controls

TD-er committed:
[Ext. RTC] Use Adafruit library to support DS1307/DS3231/PCF8523

TD-er committed:
[RTClib] Add lib/Adafruit_RTClib

TD-er committed:
[Build] Add TinyWireM to lib_ignore to prevent build issues.

TD-er committed:
[RTClib] Remove TinyWireM from depends

TD-er committed:
[Build] Lib_ignore TinyWireM for all envs

TD-er committed:
[Build] Do not use PIO library finder for RTClib (TinyWireM issues)

TD-er committed:
[Ext. RTC] Fix reading from RTC module

TD-er committed:
[Ext. RTC] Show time source + I2C scan result

TD-er committed:
[Ext. RTC] Fix some library bugs and add some validity checks

TD-er committed:
Fix TSL2591 overflow at high Lux values

See [forum](https://www.letscontrolit.com/forum/viewtopic.php?f=5&t=8652&p=53935#p53935)

TD-er committed:
[Ext.RTC] Make sure time is set immediately when received from ext. src

TD-er committed:
[Ext. RTC] Document external RTC chip support

TD-er committed:
Merge branch 'mega' into feature/DS1307_RTC

Commit:
added prometheus exporter on /metrics path

shows basic info (load, wifi strength, free memory etc) and device values

Commit:
added prometheus exporter on /metrics path

shows basic info (load, wifi strength, free memory etc) and device values

fixes from PR

Tall-Paul committed:
fix for building esp32 firmware from directory with space in name

TD-er committed:
Merge pull request #3709 from TD-er/feature/DS1307_RTC

[DS1307] Add support for external RTC clock DS1307

Tall-Paul committed:
fixes from PR

Tall-Paul committed:
Merge branch 'feature-add-prometheus-exporter' of https://github.com/Tall-Paul/ESPEasy into feature-add-prometheus-exporter

Tall-Paul committed:
Merge branch 'mega' into feature-add-prometheus-exporter

Tall-Paul committed:
more PR fixes

Tall-Paul committed:
Merge branch 'feature-add-prometheus-exporter' of https://github.com/Tall-Paul/ESPEasy into feature-add-prometheus-exporter

tonhuisman committed:
[Sysvars] Add %dns1% and %dns2% sysvars and urlencode stringfunction

tonhuisman committed:
[Sysvars] Correct typo in documentation

TD-er committed:
Merge pull request #3712 from tonhuisman/feature/add-urlencode-function-and-dns1-dns2-sysvars

[Sysvars] Add %dns1% and %dns2% sysvars and urlencode stringfunction

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p053-add-support-for-pms5003st-pms2003-pms3003

tonhuisman committed:
[P053] Only generate events when Rules enabled

tonhuisman committed:
[P053] Documentation (partly transplanted from the wiki)

TD-er committed:
[TimeSource] Move ExtTimeSource_e to TimeSource.h to fix includes

tonhuisman committed:
[P053] Add count of particles per 0.1L options to plugin, update documentation

TD-er committed:
Merge pull request #3713 from TD-er/build/missing_includes

[TimeSource] Move ExtTimeSource_e to TimeSource.h to fix includes

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p053-add-support-for-pms5003st-pms2003-pms3003

tonhuisman committed:
[P053] Correct unit terminology, improvements and fixes

tonhuisman committed:
[P053] Update documentation

tonhuisman committed:
[P053] Correct typos in the measurement units

tonhuisman committed:
[UI] Show release date in header

TD-er committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

TD-er committed:
Merge pull request #3393 from TD-er/feature/Playground_P168_ThermoOLED

[Playground] Move P168_ThermoOLED.ino to main repo

TD-er committed:
[P109] Fix issue with SPIFFS vs LittleFS

TD-er committed:
[P109] Uncrustify ThermoOLED plugin

TD-er committed:
[P109] Remove it from "display" build due to build size

TD-er committed:
[P109] Disabled also for "testing B" builds due to size.

TD-er committed:
Merge pull request #3718 from TD-er/bugfix/P109_SPIFFS

[P109] Fix issue with SPIFFS vs LittleFS

TD-er committed:
[Prometheus] Add text/plain as header

tonhuisman committed:
[P053] Limit disabling new features to 1M OTA builds

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p053-add-support-for-pms5003st-pms2003-pms3003

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/show-release-date-in-header

tonhuisman committed:
[UI] Allow Build date/name in either Header or Footer

tonhuisman committed:
[UI] Only show Build in footer

tonhuisman committed:
[P053] Correct reading from PMS2003/PMS3003, minor improvements

tonhuisman committed:
[ESP32] Fix maximum deepsleep time for ESP32

tonhuisman committed:
[P053] Minor improvements

tonhuisman committed:
[UI] Show build date in header, add build date to build filename in footer

TD-er committed:
Merge pull request #3720 from tonhuisman/bugfix/esp32-deepsleep-max-duration

[ESP32] Fix maximum deepsleep time for ESP32

tonhuisman committed:
[UI] Build name/date is link to Github Releases page

tonhuisman committed:
[UI] Open Releases link in new tab, show tooltip on hover

tonhuisman committed:
[UI] Small source improvement

tonhuisman committed:
Merge branch 'mega' into feature/show-release-date-in-header

tonhuisman committed:
[UI] Fix compiler warning

kretzp committed:
[Atlas EZO EC pH ORP] added device name to I2C address scan, reorganized plugin in Custom build

kretzp committed:
Commented Atlas EZO Plugin in Custom sample

tonhuisman committed:
[TaskValueSet] Only allowed for Dummy Device (bugfix)

tonhuisman committed:
[UI] Move Github Releases link define to ESPEasyDefaults.h

TD-er committed:
[Custom build] Make external RTC optional via USE_EXT_RTC define

Saves almost 5k in build size when not used.

tonhuisman committed:
[TaskValueSet] Rename and move new support function to Plugin.h/cpp

tonhuisman committed:
[TaskValueSet] Remove not needed include

TD-er committed:
Merge pull request #3726 from tonhuisman/bugfix/taskvalueset-only-for-dummy-device

[TaskValueSet] Only allowed for Dummy Device (bugfix)

TD-er committed:
Merge pull request #3725 from kretzp/feature/AtlasEZOPlugins_pH_EC_ORP

[Atlas EZO EC pH ORP] added device name to I2C address scan, reorgani..

TD-er committed:
Merge branch 'mega' into build/Exclude_RTC_small_builds

TD-er committed:
Merge pull request #3727 from TD-er/build/Exclude_RTC_small_builds

[Custom build] Make external RTC optional via USE_EXT_RTC define

TD-er committed:
Merge pull request #3710 from Tall-Paul/fix-post-esp32

fix for building esp32 firmware from directory with space in name

TD-er committed:
Merge branch 'mega' into feature/show-release-date-in-header

TD-er committed:
Merge pull request #3714 from tonhuisman/feature/show-release-date-in-header

[UI] Show build (date) in footer (or header)

TD-er committed:
Merge pull request #3707 from tonhuisman/feature/p053-add-support-for-pms5003st-pms2003-pms3003

[P053] Add support for PMS5003ST, PMS2003 and PMS3003

TD-er committed:
Merge branch 'mega' into bugfix/trigonometric-functions-in-normal-1M

TD-er committed:
Merge pull request #3681 from tonhuisman/bugfix/trigonometric-functions-in-normal-1M

[Build] Include Trigonometric functions also in 1M Normal builds

Tall-Paul committed:
Merge branch 'letscontrolit:mega' into feature-add-prometheus-exporter

Tall-Paul committed:
Added prometheus documentation

Tall-Paul committed:
Added ifdef, updated image name as per PR comments

Tall-Paul committed:
removed unused string definition

TD-er committed:
Merge pull request #3711 from Tall-Paul/feature-add-prometheus-exporter

added prometheus exporter on /metrics path

TD-er committed:
Merge branch 'mega' into cleanup/C014_Homie

tonhuisman committed:
[P033] Return true on PLUGIN_INIT so TaskEnable returns OK

tonhuisman committed:
[Task commands] Check if task enabled before action

tonhuisman committed:
[PLUGIN_INIT] Fix remaining plugins missing the PLUGIN_INIT case

TD-er committed:
Merge pull request #3729 from tonhuisman/bugfix/p033-return-true-on-plugin_init

[PLUGIN_INIT] Return true on PLUGIN_INIT so TaskEnable returns OK

TD-er committed:
Rename getDevicePluginID_from_TaskIndex to getPluginID_from_TaskIndex

Function name a bit more in line with similar functions.

TD-er committed:
[Commands] Output proper feedback on command execution

TD-er committed:
[Commands] Add feedback on other task commands

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into bugfix/task-commands-check-enabled

tonhuisman committed:
[Commands] Fix camelcase typo

tonhuisman committed:
[Commands] Change OK, FAILED and new error messages to uppercase

tonhuisman committed:
[Commands] Revert Failed to original case

TD-er committed:
[Commands] Don't use newline in JSON value

TD-er committed:
Merge pull request #3730 from tonhuisman/bugfix/task-commands-check-enabled

[Task commands] Check if task enabled before action (bugfix)

TD-er committed:
automatically updated release notes for mega-20210802

TD-er committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

tonhuisman committed:
[I2C Scanner] Show available device(s) for detected addresses

tonhuisman committed:
[I2C Scanner] Fix typo (swapped lines)

tonhuisman committed:
[P007] Add easier I2C Address and port selection options

tonhuisman committed:
[P007] Correct off-by-1 error for Port setting

giig1967g committed:
Fixed #3736

Longpress for PCF8574 was working only once.
Longpress timer was not reset.

tonhuisman committed:
[P007] Add documentation

tonhuisman committed:
[P007] Avoid unneeded flash-string

tonhuisman committed:
[I2C Scanner] Reworked to minimize use of Strings

tonhuisman committed:
[P103] Moved I2C address setting to designed plugin event

tonhuisman committed:
[P109] Moved I2C address setting to designed plugin event, cleanup to use macros where appropriate

tonhuisman committed:
[I2C Scanner] Restore leaving most of feature out of OTA builds

tonhuisman committed:
[I2C Scanner] Show proper message if no I2C pins configured

tonhuisman committed:
[I2C Scanner] Adjust errormessage, refactor checks into isI2CEnabled()

tonhuisman committed:
[P103] Remove unsupported devices from I2C selection list

tonhuisman committed:
[P103] Fix silly typo :-(

tonhuisman committed:
[P108] DDS238 Fix 2-complement values (#3662)

giig1967g committed:
Updated plugin documentation

clumsy-stefan committed:
Exclude not needed parts for USES_MQTT and WEBSERVER_METRICS

Exclude not needed code to save space (USES_MQTT) and make it compile again (WEBSERVER_METRICS).

tonhuisman committed:
[P108] Code improvements

tonhuisman committed:
[I2C Scanner] Move refactored check into Settings struct

TD-er committed:
[TaskValueSet] Allow TaskValueSet on non-Dummy tasks (setting)

As reported [here](https://github.com/TD-er/ESPEasy/commit/92f9006de6710c6b21eb71d60b089e7d9ed966be#r54477663) by @clumsy-stefan

TD-er committed:
Merge pull request #3737 from giig1967g/bugfix/3736

Fixed #3736

TD-er committed:
Merge pull request #3740 from tonhuisman/bugfix/p108-correct-2-complement-values

[P108] DDS238 Fix 2-complement values (#3662)

clumsy-stefan committed:
Cleanup webserver_metrics

Remove unneeded lines

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/i2c-scanner-show-plugin-name

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/P065-add-mode-and-repeat-commands

tonhuisman committed:
Merge branch 'feature/P065-add-mode-and-repeat-commands' of https://github.com/tonhuisman/ESPEasy-1 into feature/P065-add-mode-and-repeat-commands

TD-er committed:
Merge pull request #3742 from clumsy-stefan/fix_metrics_build

Exclude not needed parts for USES_MQTT and WEBSERVER_METRICS

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/P065-add-mode-and-repeat-commands

tonhuisman committed:
[P065] Adjust documentation on mode command not always working as advertised

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/i2c-scanner-show-plugin-name

tonhuisman committed:
[I2C Scanner] Also check I2C Low Speed (is used by scanner)

TD-er committed:
[Hostname] Make generated hostname RFC952 compliant (#3723)

Fixes: #3723

TD-er committed:
[Cleanup] Use Arduino isDigit and isAlphaNumeric functions

TD-er committed:
Merge pull request #3733 from tonhuisman/feature/i2c-scanner-show-plugin-name

[I2C Scanner] Show available device(s) for detected addresses

TD-er committed:
Merge pull request #3734 from tonhuisman/feature/p007-add-easier-address-and-port-selection

[P007] PCF8591 Add easier I2C Address and port selection options

TD-er committed:
[Docs] Document  "Allow TaskValueSet on all plugins" checkbox

TD-er committed:
Merge branch 'mega' into feature/taskvalueset_all_plugins

TD-er committed:
[Formula] Use correct nr of decimals in formula (#3721)

Fixes: #3721

TD-er committed:
[Formula] Allow using standard conversions and task values in formula

TD-er committed:
[Formula] Add more documentation for formulas + examples

TD-er committed:
[docs] Fix wrong topic length in docs.

TD-er committed:
Merge pull request #3745 from TD-er/bugfix/format_decimals_formula

[Formula] Use correct nr of decimals in formula (#3721)

TD-er committed:
Merge branch 'mega' into cleanup/C014_Homie

TD-er committed:
[Cleanup] Replace single character flash strings

tonhuisman committed:
Merge branch 'mega' into feature/P065-add-mode-and-repeat-commands

TD-er committed:
Merge pull request #3615 from tonhuisman/feature/P065-add-mode-and-repeat-commands

[P065] Add mode and repeat commands

TD-er committed:
Merge pull request #3744 from TD-er/bugfix/RFC952_hostname

[Hostname] Make generated hostname RFC952 compliant (#3723)

TD-er committed:
Merge pull request #3743 from TD-er/feature/taskvalueset_all_plugins

[TaskValueSet] Allow TaskValueSet on non-Dummy tasks (setting)

TD-er committed:
Merge pull request #3708 from TD-er/cleanup/C014_Homie

Cleanup C014 Homie

tonhuisman committed:
[RTClib] (Bugfix) Write month correctly to DS3231 (#3749)

TD-er committed:
Merge pull request #3751 from tonhuisman/bugfix/rtc-ds3231-write-month-correctly

[RTClib] (Bugfix) Write month correctly to DS3231 (#3749)

TD-er committed:
Fix setting month + century bit RTC lib

As suggested by @uzi18

TD-er committed:
Merge pull request #3752 from letscontrolit/bugfix/RTClib_month

Fix setting month + century bit RTC lib

Rini52 committed:
Merge branch 'mega' of https://github.com/Rini52/ESPEasy into mega

TD-er commented:
This PR cannot be merged.
It contains over 200 commits and over 190 files changed.
